File: safari-534.53-branch/LayoutTests/http/tests/navigation/resources/no-referrer-helper.php
<?php
    // Prevent from being cached.
    header("Cache-Control: no-store, private, max-age=0");
    header("Content-Type: text/html");
?>

<html><body>
<div id="console"></div>
<div id="referrer">
Referrer: <?php echo $_SERVER['HTTP_REFERER']; ?>
</div>
<br/>
<script>
function log(msg)
{
    var line = document.createElement('div');
    line.appendChild(document.createTextNode(msg));
    document.getElementById('console').appendChild(line);
}

    var consoleWindow = window.open("", "consoleWindow");
    if (consoleWindow) {
        consoleWindow.log(document.getElementById("referrer").innerText);
        consoleWindow.log("window.opener: " + (window.opener ? window.opener.location : ""));
    }
    
    if (window.layoutTestController)
        layoutTestController.notifyDone();
</script>

</body></html>

File: LayoutTests/http/tests/navigation/resources/no-referrer-helper.py
#!/usr/bin/env python3

import os
import sys

sys.stdout.write(
    'Cache-Control: no-store, private, max-age=0\r\n'
    'Content-Type: text/html\r\n\r\n'
)

print('''<html><body>
<div id="console"></div>
<div id="referrer">
Referrer: {}
</div>
<br/>
<script>
function log(msg)
{{
    var line = document.createElement('div');
    line.appendChild(document.createTextNode(msg));
    document.getElementById('console').appendChild(line);
}}

    console.log(document.getElementById("referrer").innerText);
    console.log("window.opener: " + (window.opener ? window.opener.location : ""));

    if (window.testRunner)
        testRunner.notifyDone();
</script>

</body></html>'''.format(os.environ.get('HTTP_REFERER', '')))
